components/gigya/globalEvents/src/index.js

```diff
@@ -8,34 +8,24 @@ import {
   removeOnLogoutSubscriber
 } from './helper'
 
-class GigyaGlobalEvents extends Component {
-  _registerEvents = ({onLogin, onLogout}) => {
+export default function GigyaGlobalEvents({onLogin, onLogout}) {
+  const registerEvents = ({onLogin, onLogout}) => {
     onLogin && addOnLoginSubscriber(onLogin)
     onLogout && addOnLogoutSubscriber(onLogout)
   }
 
-  _unregisterEvents = ({onLogin, onLogout}) => {
+  const unregisterEvents = ({onLogin, onLogout}) => {
     onLogin && removeOnLoginSubscriber(onLogin)
     onLogout && removeOnLogoutSubscriber(onLogout)
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line
-    this._unregisterEvents(this.props)
-    this._registerEvents(nextProps)
-  }
-
-  UNSAFE_componentWillMount() { // eslint-disable-line
+  useEffect(function() {
     initListener()
-    this._registerEvents(this.props)
-  }
-
-  componentWillUnmount() {
-    this._unregisterEvents(this.props)
-  }
+    registerEvents({onLogin, onLogout})
```

Comment:
Maybe you can move the functions `registerEvents` and `unregisterEvents` inside the useEffect. 
Didn't you skip the linter or have you wanted to ignore the deps of useEffect in an intentional way?

Reply:
I'm just keeping the functionality of before that was using the `componentWillReceiveProps` without any check in order to register again the events.